tests/ci/cdk/cdk/windows_docker_image_build_stack.py

```diff
@@ -119,6 +130,11 @@ def __init__(self, scope: Construct, id: str, **kwargs) -> None:
             vpc_subnets=ec2.SubnetSelection(subnet_type=ec2.SubnetType.PUBLIC),
             machine_image=machine_image,
             user_data=setup_user_data,
+            instance_name="{}-instance".format(id),
         )
 
         Tags.of(instance).add(WIN_EC2_TAG_KEY, WIN_EC2_TAG_VALUE)
+
+        self.output = {
+            "s3_bucket_name": bucket._generate_physical_name(),
```

Comment:
Functions prefixed by `_` while not completely hidden, unlike `__`, are still considered private and it's bad from to be using such functions. You should formulate an alternative unless you have documentation from the AWS CDK that states it can be used.